applicationset/generators/generator_spec_processor.go

```diff
@@ -31,7 +33,17 @@ func Transform(requestedGenerator argoprojiov1alpha1.ApplicationSetGenerator, al
 			}
 			continue
 		}
-
+		if len(genParams) != 0 {
+			err := interpolateGenerator(&requestedGenerator, genParams)
+			if err != nil {
+				log.WithError(err).WithField("genParams", genParams).
+					Error("error interpolating params for generator")
+				if firstError == nil {
+					firstError = err
+				}
+				continue
+			}
+		}
 		params, err := g.GenerateParams(&requestedGenerator, appSet)
```

Comment:
in addition to above change replace requestedGenerator with newrequestedgenerator